src/Decompiler/Loading/Loader.cs

```diff
@@ -298,6 +309,30 @@ public Program LoadRawImage(ImageLocation imageLocation, byte[] image, Address?
             return program;
         }
 
+        private byte[]? ExtractRawFileFromArchives(ImageLocation imageLocation, byte[] image, Address? addrLoad)
+        {
+            var fsLocation = ImageLocation.FromUri(imageLocation.FilesystemPath);
+            var binaryImage = LoadBinaryImage(fsLocation, image, null, addrLoad);
+            for (int i = 0; i < imageLocation.Fragments.Length; ++i)
+            {
+                var fragment = imageLocation.Fragments[i];
+                if (binaryImage is not IArchive archive)
+                    throw new InvalidOperationException($"Image '{imageLocation}' expects an archive or file container, but none was loaded.");
+                var item = archive[fragment];
+                if (item is not ArchivedFile file)
+                    throw new InvalidOperationException($"Fragment '{fragment}' does not refer to an archived file.");
+                binaryImage = file.LoadImage(Services, addrLoad);
+                if (binaryImage is Blob blob)
+                {
+                    if (i == imageLocation.Fragments.Length - 1)
+                        return blob.Image;
```

Comment:
What about case (rare case) when file format is unknown to `Reko` but user overrides it by `Open As` dialog?
I suggest move this check before loading image from archive file: https://github.com/uxmal/reko/pull/1098/commits/02db8bf1742b68f2cfb5fa4a8f35806bdd8a6e8d#r761658787